src/ORM/DataObject.php

```diff
@@ -1295,6 +1303,32 @@ protected function onBeforeWrite()
      */
     protected function onAfterWrite()
     {
+        // Fetch values from any generated columns
+        // since we may have updated the columns they are based on
+        $skipGeneratedColumns = static::config()->get('skip_afterwrite_generatedcolumn_fetch');
+        if ($skipGeneratedColumns !== true) {
+            $schema = DataObject::getSchema();
+            $generatedFields = array_keys($schema->generatedFields(static::class));
+            if (is_array($skipGeneratedColumns)) {
+                $generatedFields = array_diff($generatedFields, $skipGeneratedColumns);
+            }
+            // Only fetch if we have generated columns that aren't skipped
+            if (!empty($generatedFields)) {
+                $select = static::get()
+                    ->filter(['ID' => $this->ID])
+                    ->sort(null)
+                    ->limit(1)
+                    ->dataQuery()
+                    ->getFinalisedQuery();
+                $columns = [];
+                foreach ($generatedFields as $fieldName) {
+                    $columns[] = $schema->sqlColumnForField(static::class, $fieldName);
+                }
+                $select->setSelect($columns);
+                $this->update($select->execute()->record());
+            }
+        }
```

Comment:
Without this change, the old value will display after saving the record. You wouldn't see the new generated value until you refresh.

This will fire a single optimised `SELECT` query (which only fetches the generated columns, nothing else) any time a record is saved - but _only_ if that record has generated columns.
Records with no generated columns won't have this additional query.